presto-hive/src/test/java/io/prestosql/plugin/hive/acid/TestOrcAcidPageSource.java

```diff
@@ -0,0 +1,230 @@
+    {
+        assertRead(ImmutableSet.copyOf(NationColumn.values()), OptionalInt.of(100));
+    }
+
+    /**
+     * Tests stripe stats and row groups stats based pruning works fine
+     */
+    @Test
+    public void testSomeStripesAndRowGroupRead()
+    {
+        assertRead(ImmutableSet.copyOf(NationColumn.values()), OptionalInt.of(5));
+    }
+
+    private static void assertRead(Set<NationColumn> columns, OptionalInt nationKeyPredicate)
+    {
+        TupleDomain<HiveColumnHandle> tupleDomain = TupleDomain.all();
+        if (nationKeyPredicate.isPresent()) {
+            tupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(toHiveColumnHandle(NATION_KEY), Domain.singleValue(INTEGER, (long) nationKeyPredicate.getAsInt())));
+        }
+
+        List<Nation> actual = readFile(columns, tupleDomain);
+
+        List<Nation> expected = new ArrayList<>();
+        for (Nation nation : ImmutableList.copyOf(new NationGenerator().iterator())) {
+            if (!nationKeyPredicate.isPresent() || nationKeyPredicate.getAsInt() == nation.getNationKey()) {
+                expected.addAll(nCopies(1000, nation));
+            }
+        }
+
+        assertNations(columns, actual, expected);
+    }
+
+    private static List<Nation> readFile(Set<NationColumn> columns, TupleDomain<HiveColumnHandle> tupleDomain)
+    {
+        List<HiveColumnHandle> columnHandles = columns.stream()
+                .map(TestOrcAcidPageSource::toHiveColumnHandle)
+                .collect(toImmutableList());
+
+        List<String> columnNames = columnHandles.stream()
+                .map(HiveColumnHandle::getName)
+                .collect(toImmutableList());
+
+        ConnectorPageSource pageSource = PAGE_SOURCE_FACTORY.createPageSource(
+                new JobConf(new Configuration(false)),
+                SESSION,
+                new Path(TEST_FILE.getAbsoluteFile().toURI()),
+                0,
+                TEST_FILE.length(),
+                TEST_FILE.length(),
+                createSchema(),
+                columnHandles,
+                tupleDomain,
+                DateTimeZone.UTC).get();
+
+        int nationKeyColumn = columnNames.indexOf("n_nationkey");
+        int nameColumn = columnNames.indexOf("n_name");
+        int regionKeyColumn = columnNames.indexOf("n_regionkey");
+        int commentColumn = columnNames.indexOf("n_comment");
+
+        ImmutableList.Builder<Nation> rows = ImmutableList.builder();
+        while (!pageSource.isFinished()) {
+            Page page = pageSource.getNextPage();
+            if (page == null) {
+                continue;
+            }
+
+            page = page.getLoadedPage();
+            for (int position = 0; position < page.getPositionCount(); position++) {
+                long nationKey = -1;
+                if (nationKeyColumn >= 0) {
+                    nationKey = INTEGER.getLong(page.getBlock(nationKeyColumn), position);
+                }
+
+                String name = "INVALID";
```

Comment:
`"<not read>"` to better distinguish it from any other potential source of the text

same for -1 for numeric columns -- eg `-42` is less likely to come from a file than -1

Reply:
Changing. But the values in the current test's file are guaranteed to be not INVALID or -1.